pkg/daemon/server/service/rater/options.go

```diff
@@ -28,14 +28,15 @@ type Option func(*options)
 func defaultOptions() *options {
 	// A simple example of how these numbers work together:
 	// Assuming we have 200 tasks, we have 20 workers, each worker will be responsible for approximately 10 tasks during one iteration.
-	// The task interval is 10 seconds, which means each task need to be picked up by a worker every 10 seconds.
-	// Hence a worker needs to finish processing 1 task in 1 second.
-	// Translating to numaflow language, for a 200-pod pipeline, a worker needs to finish scraping 1 pod in 1 second, which is a reasonable number.
-	// TODO - need to adjust the numbers once we have a better idea of the performance
+	// The task interval is 5 seconds, which means each task need to be picked up by a worker every 5 seconds.
```

Comment:
How do we know 5 seconds is a better number?

Reply:
We use a 10s time window to collect count metrics. To guarantee that we include ALL pods in each of the 10s window, we need to ensure each pod gets picked up and scrapped every 5 seconds(10/2). I won't say it's a better number. It's our current configuration which is tested to be able to handle 100-pod vertices' use case.

Reply:
even though we show wrong rates, it will converge if two consecutive fetches succeed.

Reply:
The way we calculate rates is that we calculate total delta by summing up all deltas among each of the consecutive TimestampedCounts pairs within lookback seconds. If the lookback second is 300s, there will be ~30 deltas calculated. If we have a pod wrongly missed in one of the TimestampedCounts, the calculated rate for lookback 300s will continue being off until that TimestampedCounts passes the 300s timeline, meaning the full converge takes 300 seconds.

Reply:
fair point